### DISCUSSION TITLE
Only 2 of 7 speakers visible??

### DISCUSSION THREAD

**Discussion by udouble** [music-assistant/support#1603] [Q&A] [answered]
I have...
- Bose Soundtouch integration 
- DLNA Digital Media Renderer
- Spotify integration
- your lovelace card
... correctly installed. 

I have 7 speakers in 7 different rooms:

![Scherm­afbeelding 2023-11-03 om 19 21 40](https://github.com/music-assistant/hass-music-assistant/assets/15670896/ec3b6b10-f166-486a-b53d-b0c32da4d865)

Still, I can only choose two speakers:

![Scherm­afbeelding 2023-11-03 om 19 22 37](https://github.com/music-assistant/hass-music-assistant/assets/15670896/9268f7fe-0f65-4f12-9fbe-8b5c2f9c36e3)

...while i would like to choose all of them and even play different music in different rooms. 

Any suggestions?

**Comment by OzGav** [comment]
It doesn’t look like you have the HA integration installed as none of those speakers you show are being provided by MA. You need to install that from HACS. Please read the docs on GitHub

**Reply by udouble** [reply]
I did - maybe i was filtering the speakers. But I'm removing it, since it is too unstable. Thanks anyway. The Spotify lovelace card is solving my problem.

**Reply by OzGav** [reply, marked as answer]
It doesn't look like you have but if all you need is Spotify then no problems
